tests/test_python.py

```diff
@@ -109,21 +110,36 @@ def test_tox(auto_container):
     auto_container.connection.run_expect([0], "pip install --user tox")
 
 
-def test_python_devel(auto_container):
-    """Check that python_devel package can be installed."""
+def test_cryptography(auto_container):
+    """Check that cryptography python module can be installed from source so that
+    it is built against the SLE BCI provided openssl rather than an upstream copy."""
     version = auto_container.connection.run_expect(
         [0], "echo $PYTHON_VERSION"
     ).stdout.strip()
-    if "3.6" in version:
-        pkg = "python3-devel"
-    elif "3.9" in version:
-        pkg = "python39-devel"
-    elif "3.10" in version:
-        pkg = "python310-devel"
-    else:
-        raise Exception(f"Unknown python version: {version}")
+
+    if packaging.version.Version(version) < packaging.version.Version("3.8"):
+        pytest.skip("cryptography tests only supported on >= 3.8")
+
+    # install dependencies
+    auto_container.connection.run_expect(
+        [0],
+        f"zypper --non-interactive in cargo libffi-devel openssl-devel gcc tar gzip",
+    )
+
+    # install cryptography from source
     auto_container.connection.run_expect(
-        [0], f"zypper --non-interactive in {pkg}"
+        [0], f"pip install --no-binary :all: cryptography"
+    )
+
+    cryptography_version = "38.0.3"
```

Comment:
This does not look like a good idea to me to be honest. No one will remember to bump this version :-/

Reply:
I share Dan's concerns here, especially with cryptographic tests this is dangerous.

Reply:
there is no need to bump the version. the reason for hardcoding the version is that the test doesn't break with the release of new versions of cryptography that have new dependencies.

Reply:
Our users will most likely use the most recent version and not whatever we pinned today. Thus there is imho very much a need to bump the version.